segmentstore/server/src/test/java/io/pravega/segmentstore/server/logs/TestLogItem.java

```diff
@@ -57,37 +63,49 @@ public void failSerializationAfterComplete(double ratio, IOException exception)
         this.exception = exception;
     }
 
-    public byte[] getData() {
-        return this.data;
+    @SneakyThrows(IOException.class)
+    byte[] getFullSerialization() {
+        byte[] result = new byte[Long.BYTES + Integer.BYTES + this.data.length];
+        serialize(new FixedByteArrayOutputStream(result, 0, result.length));
+        return result;
     }
 
-    public byte[] getFullSerialization() {
-        byte[] result = new byte[Long.BYTES + Integer.BYTES + this.data.length];
-        try {
-            this.serialize(new FixedByteArrayOutputStream(result, 0, result.length));
-        } catch (IOException ex) {
-            throw new RuntimeException(ex);
+    @Override
+    public void serialize(OutputStream output) throws IOException {
+        DataOutputStream dataOutput = new DataOutputStream(output);
+        dataOutput.writeLong(this.sequenceNumber);
+        dataOutput.writeInt(this.data.length);
+        val ex = this.exception;
+        if (ex == null) {
+            dataOutput.write(this.data);
+        } else {
+            int breakPoint = (int) (this.data.length * this.failAfterCompleteRatio);
+            dataOutput.write(this.data, 0, breakPoint);
+            throw ex;
         }
-
-        return result;
     }
 
     @Override
-    public long getSequenceNumber() {
-        return this.seqNo;
+    public int hashCode() {
+        return Long.hashCode(this.sequenceNumber);
     }
 
     @Override
-    public void serialize(OutputStream output) throws IOException {
-        DataOutputStream dataOutput = new DataOutputStream(output);
-        dataOutput.writeLong(seqNo);
-        dataOutput.writeInt(data.length);
-        if (this.exception == null) {
-            dataOutput.write(data);
-        } else {
-            int breakPoint = (int) (data.length * this.failAfterCompleteRatio);
-            dataOutput.write(data, 0, breakPoint);
-            throw this.exception;
+    public boolean equals(Object obj) {
+        if (obj instanceof TestLogItem) {
+            TestLogItem other = (TestLogItem) obj;
+            boolean match = getSequenceNumber() == other.getSequenceNumber() && this.data.length == other.data.length;
+            if (match) {
+                for (int i = 0; i < this.data.length; i++) {
+                    if (this.data[i] != other.data[i]) {
+                        return false;
```

Comment:
Yes... At line 97 it does a SeqNo COmparison. Here it does an index-by-index comparison between the two items' data.